src/main/clojure/clara/rules.cljc

```diff
@@ -410,3 +410,21 @@ See the [query authoring documentation](http://www.clara-rules.org/docs/queries/
            (cond-> ~(dsl/parse-query* binding definition {} (meta &form))
              ~name (assoc :name ~(str (clojure.core/name (ns-name *ns*)) "/" (clojure.core/name name)))
              ~doc (assoc :doc ~doc)))))))
+
+#?(:clj
+   (defmacro clear-ns-productions!
+     "Ensures that any rule/query definitions which have been cached will be cleared from the associated namespace.
+      Rule and query definitions can be cached such if their definitions are not explicitly overwritten with the same
```

Comment:
nitpick - Did you mean "such that" here?

Reply:
Yes.

Reply:
Looks good to me overall.